web/src/hash_util.ts

```diff
@@ -216,27 +225,18 @@ export function validate_stream_settings_hash(hash: string): string {
         //
         // In all these cases we redirect the user to 'subscribed' tab.
         if (sub === undefined || (page_params.is_guest && !stream_data.is_subscribed(stream_id))) {
-            return "#streams/subscribed";
+            return channels_settings_section_url();
         }
 
-        const stream_name = hash_components[2];
         let right_side_tab = hash_components[3];
         const valid_right_side_tab_values = new Set(["general", "personal", "subscribers"]);
-        if (sub.name === stream_name && valid_right_side_tab_values.has(right_side_tab)) {
```

Comment:
This also did not account for a stream name not being encoded in the hash/URL.

Reply:
Ohh yes, thanks for fixing this.